pandas/tests/scalar/timestamp/test_timestamp.py

```diff
@@ -1092,3 +1092,17 @@ def test_constructor_ambigous_dst():
     expected = ts.value
     result = Timestamp(ts).value
     assert result == expected
+
+
+def test_constructor_before_dst_switch():
+    # GH 31043
+    # Make sure that calling Timestamp constructor
+    # on time just before DST switch doesn't lead to
+    # nonexistent time
+    epoch = 1552211999999999872
+    ts = Timestamp(epoch, tz="dateutil/US/Pacific")
+    delta = ts.replace(tzinfo=None) - datetime.utcfromtimestamp(0)
+    assert delta.total_seconds() < 1552183200
```

Comment:
Expanded the test a bit to see whether we end up rounding after all on most environments or if something else is going on.